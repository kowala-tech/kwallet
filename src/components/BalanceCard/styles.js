const styles = (theme) => ({
	root: {
		marginBottom: theme.spacing.unit * 2
	},
	text: {
		color: "#fff",
		textAlign: "center",
		fontWeight: 200,
		paddingBottom: theme.spacing.unit
	},
	balance: {
		color: "#fff",
		textAlign: "center",
		fontWeight: 400,
		paddingBottom: theme.spacing.unit * 2
	},
	balanceLabel: {
		color: "#fff",
		textAlign: "center",
		fontWeight: 400,
		paddingBottom: theme.spacing.unit * 2
	}
});

export default styles;
