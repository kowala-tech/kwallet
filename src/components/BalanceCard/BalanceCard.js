import React from "react";
import PropTypes from "prop-types";
import { FormattedNumber } from "react-intl";
// Material UI Imports
import { withStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
// Component Related Imports
import styles from "./styles";
import { weiToKusd } from "../../transforms/currency";

class BalanceCard extends React.Component {

	render() {
		const {
			classes,
			balance
		} = this.props;

		return (
			<div className={classes.root}>
				<Typography
					className={classes.text}
				>
					Your wallet's current balance is
				</Typography>
				<Typography
					variant="title"
					className={classes.balance}
				>
					<FormattedNumber
						minimumFractionDigits={4}
						maximumFractionDigits={18}
						value={weiToKusd(balance)}
					/>
					{" kUSD"}
				</Typography>
			</div>
		);
	}
}

BalanceCard.propTypes = {
	classes: PropTypes.object.isRequired,
	balance: PropTypes.string
};

export default withStyles(styles)(BalanceCard);
