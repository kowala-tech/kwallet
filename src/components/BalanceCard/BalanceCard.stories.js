import React from "react";
import { storiesOf } from "@storybook/react";
// Story related component imports
import BalanceCard from "./BalanceCard";

const ContrastBackground = (storyFn) => (
	<div style={{ backgroundColor: "#1f033d" }}>
		{ storyFn() }
	</div>
);

storiesOf("Balance Card", module)
	.addDecorator(ContrastBackground)
	.add("with balance", () => (
		<BalanceCard
			balance={12.3456789}
		/>
	));
