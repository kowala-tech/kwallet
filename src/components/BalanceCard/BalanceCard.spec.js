import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
// Story related component imports
import BalanceCard from "./BalanceCard";

describe("BalanceCard", () => {

	it("it should match the snapshot", () => {
		const component = toJson(
			shallow(
				<BalanceCard
					balance={"43.23438"}
				/>
			)
		);
		expect(component).toMatchSnapshot();
	});

});
