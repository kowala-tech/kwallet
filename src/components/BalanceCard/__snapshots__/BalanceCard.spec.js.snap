// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BalanceCard it should match the snapshot 1`] = `
<BalanceCard
  balance="43.23438"
  classes={
    Object {
      "balance": "BalanceCard-balance-3",
      "balanceLabel": "BalanceCard-balanceLabel-4",
      "root": "BalanceCard-root-1",
      "text": "BalanceCard-text-2",
    }
  }
/>
`;
